articles/language/statements.md

```diff
@@ -149,45 +150,25 @@ Q# distinguishes between mutable and immutable symbols.
 In general, the use of immutable symbols is encouraged because it
 allows the compiler to perform more optimizations.
 
-For arrays, mutability or immutability applies both to the array as a whole
-and to array elements.
-That is, the elements of an immutable array may not be changed.
-
-The contents of a tuple instance or of a user-defined type based on a tuple
-may not be changed regardless of the mutability of the symbol bound to the
-instance.
-If a mutable symbol is bound to a tuple, the symbol may be bound
-to a new tuple, but the existing tuple may not be modified in place.
+The left-hand-side of the binding consists of a symbol tuple, and the right hand side of an expression.
 
-The arguments to an operation or a function are treated as immutable.
-This also applies to elements of an array passed to an operation or function.
-This means that Q# does not support "out" variables.
+Since all Q# types are value types - with the qubits taking a somewhat special role - 
+formally a "copy" is created when a value is bound to a symbol, or when a symbol is rebound. 
```

Comment:
It feels weird to denote something as formal and then immediately follow up with quote marks. If I understand right, the word "formal" here refers to the mathematical formalism that one could write for Q# semantics if they felt so inclined, rather than a formal description of how those semantics are implemented by the Q# compiler.
As you've noted here and elsewhere, we intentionally don't define when a copy of a value is made _in practice_, since memory locations are not directly observable within Q#.

The following lines help clarify that some, but it might be good to reword this paragraph a little to highlight what formalisms are being referred to.

Reply:
Yes, I tried to follow up with what I mean by this "formally". If that's ok, I'll leave it for now so we can get this large bunch of changes together, and then we can improve on it.